crypto/trace.c

```diff
@@ -496,6 +496,8 @@ void OSSL_trace_end(int category, BIO * channel)
     char *suffix = NULL;
 
     category = ossl_trace_get_category(category);
+    if (category < 0)
+        return;
```

Comment:
It‘s a programming error if the trace groups are not properly nested. Instead of silently ignoring it, I‘d add `ossl_assert(category >= 0)`. (Note that there is already an assertion a few lines below). 

(Disclaimer: this is a quick drive-by comment from m mobile phone)

Reply:
Hmmm… just read issue #17353. I need to take another look when I‘m back on my computer.

Reply:
We normally reserve `ossl_assert` and `assert` for catching internal errors. This is actually public API, and so the "bug" would be in the caller and therefore may not be an "internal" error. So, its not clear to me that `ossl_assert` is actually appropriate.

Reply:
It is not an *assert* context. This is something that can easily happen with a public API.